src/MutableMap/MutableMap.dfy

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT
+*******************************************************************************/
+
+// RUN: %dafny /compile:0 "%s"
+
+module {:extern "MutableMap"} {:options "/functionSyntax:4"} MutableMap {
+  class {:extern "MutableMap"} MutableMap<K(==),V(==)> {
+    function {:extern "content"} content(): map<K, V>
+      reads this
+
+    constructor {:extern "MutableMap"} ()
+      ensures this.content() == map[]
+
+    method {:extern "put"} Put(k: K, v: V)
+      modifies this
+      ensures this.content() == old(this.content())[k := v]    
+
+    function {:extern "keys"} Keys(): (keys: set<K>)
+      reads this
+      ensures keys == this.content().Keys
+
+    function {:extern "values"} Values(): (values: set<V>)
```

Comment:
Could we leave off `Values()` and `Items()` for now, so that we don't force `V` to be equality-supporting unnecessarily?

Alternatively, you could make those methods instead of functions and have them return non-deterministically ordered sequences instead of sets.

Reply:
I'm especially inclined to leave these out given https://github.com/dafny-lang/dafny/issues/1373